src/Mono.Android/Android.Runtime/JNINativeWrapper.cs

```diff
@@ -27,6 +28,11 @@ static void get_runtime_types ()
 
 		public static Delegate CreateDelegate (Delegate dlg)
 		{
+			// FIXME: https://github.com/xamarin/xamarin-android/issues/8724
+			[UnconditionalSuppressMessage ("AOT", "IL3050", Justification = "NativeAOT is not yet supported.")]
+			static DynamicMethod CreateDynamicMethod (Type ret_type, Type [] param_types) =>
```

Comment:
I must admit I'm not a fan of doing it this way. For trim related warnings we used XML files added only during the repo build. Unfortunately, that doesn't work for analyzers and NativeAOT (https://github.com/dotnet/runtime/issues/79393).

I can see two other solutions which might work reasonably well:
* Since we only care about analyzers right now (no NativeAOT support), suppress these via `#pragma warning disable` (and add a comment why we're doing it that way). That would mean that if/when somebody actually does try NativeAOT, they'll get these warnings, which is what we want.
* Suppress them in a single file - using assembly level attributes and the Target property syntax of the attribute (that "should" work even for analyzer, but I didn't play with it). That way they're at least in the same place....

Reply:
Ok, it makes sense we might not want `[UnconditionalSuppressMessage ("AOT", "IL3050"`, as it would hide the warning from anyone consuming it from a NativeAOT application.

I'll review what we can do instead.